### PR TITLE
lookback and access are no longer binops, max iter added to gn

### PR DESCRIPTION
**Pull request by j-hui**

### CODE REVIEW

File: src/backend/sast.mli [hunk 1]
```diff
@@ -4,13 +4,14 @@ type styp =
   | SString
   | SBool
   | SStruct of string
-  | SArray of styp * int option (* necessary for struct def'ns, not used elsewhere *)
+  | SArray of styp * int option
+  | Ptr
```
Comment (andylhxu) [inline src/backend/sast.mli]: leaving out `_` in the front?

Comment (j-hui) [reply src/backend/sast.mli]: what?

Comment (j-hui) [reply src/backend/sast.mli]: dude this isn't our syntax – this is just the name of an algebraic type

**Comment by j-hui** [comment]
the last PR is just to get it to compile; `codegen.ml` basically does nothing now

**Review by Contentmaudlin — APPROVED** [review]
kewl

**Pull request merged**